pkg/config/application_kubernetes.go

```diff
@@ -36,6 +36,13 @@ type KubernetesApplicationSpec struct {
 	TrafficRouting *KubernetesTrafficRouting `json:"trafficRouting"`
 	// The label will be configured to variant manifests used to distinguish them.
 	VariantLabel KubernetesVariantLabel `json:"variantLabel"`
+	// List of matching conditions to resolve the destinations for application resources.
+	// Each resource will be checked over the match conditions of each destination.
+	// If matches, it will be applied to that destination,
+	// otherwise, it will be fallen through the next destination to check.
+	// Any resource which does not match any specified destination will be applied
+	// to the default destination which had been specified while registering the application.
+	ResourceDestinations []KubernetesResourceDestination `json:"resourceDestinations"`
```

Comment:
Sorry but I started feeling that the `destination` and the `provider` concepts are mixing here. Should we rename this to `resourceProviders`?

Reply:
I get your point. We need a better name. 🤔 
@khanhtc1202 @knanao  Wdyt about `resourceRoutes` like envoy?
https://github.com/pipe-cd/pipecd/blob/master/manifests/pipecd/templates/envoy-configmap.yaml#L63-L68

Or any better names?

Reply:
> Wdyt about resourceRoutes like envoy?
I feel it's better 🙌

Reply:
I got your point, now I understand why the name was `resourceDestinations` at first 😂 

tbh, `resourceRoutes` is a fine name, people with a sense of k8s or envoy things can understand what it means, but same as `resourceDestinations` the issue of mixing what it's and what it requires does exist. Maybe, it's just my overthinking so please don't be overthinking about this 🙏 On the second thought, the relation here is: we have a place that we want to deploy the resource and the configuration contains a provider to deploy the resource to that place, which is clear enough to me 👍 

In that context, `resourceDestinations`, `resourceRoutes`, or `resourceTargets` are all fine names, maybe `resourceTargets` is a bit better (since it's the shortest).

Reply:
@knanao @khanhtc1202 Thanks for your comments.
I have just updated to use `resourceRoutes`.
In addition to your specified reasons, the configuration struct contains both `provider` (where to apply) and `match` (condition to check) so I feel that `route` is better than just `destination`.